custom/enikshay/integrations/nikshay/repeater_generator.py

```diff
@@ -219,6 +227,117 @@ def handle_exception(self, exception, repeat_record):
+        bacilli_count = test_case_properties.get('max_bacilli_count', None)
+        result_grade = self.get_result_grade(test_result_grade, bacilli_count)
+
+        if any(mandatory_value is None for mandatory_value in [lab_serial_number, result_grade]):
+            raise RequiredValueMissing("Mandatory value missing in one of the following "
+                                       "LabSerialNo: {lab_serial_number}, ResultGrade: {result_grade}"
+                                       .format(lab_serial_number=lab_serial_number,
+                                               result_grade=test_result_grade))
+
+        return interval_id, lab_serial_number, result_grade, dmc_code
+
+    def get_result_grade(self, test_result_grade, bacilli_count):
+        if test_result_grade in smear_result_grade.keys():
+            return smear_result_grade.get(test_result_grade)
+        elif test_result_grade == 'scanty':
+            return smear_result_grade.get("SC-{b_count}".format(b_count=bacilli_count), None)
+
+    def _get_interval_id(self, testing_purpose, follow_up_test_reason):
+        if testing_purpose == 'diagnostic':
+            interval_id = 0
+        else:
+            interval_id = purpose_of_testing.get(follow_up_test_reason, None)
+        if interval_id is None:
+            raise RequiredValueMissing("Value missing for intervalID, purpose_of_testing: {testing_purpose}, "
+                                       "follow_up_test_reason: {follow_up_test_reason}".format(
+                testing_purpose=testing_purpose,
+                follow_up_test_reason=follow_up_test_reason)
+            )
+        return interval_id
+
+    def _get_dmc_code(self, test_case_properties):
+        dmc_location_id = test_case_properties.get("testing_facility_id", None)
+        if not dmc_location_id:
+            # fallback to lab referral case owner id for older versions of app
+            lab_referral_case = get_lab_referral_from_test(self.test_case.domain, self.test_case.get_id)
+            dmc_location_id = lab_referral_case.owner_id
+        if not dmc_location_id:
+            raise RequiredValueMissing("Value missing for dmc_code/testing_facility_id for test case: " +
+                                       self.test_case.get_id)
+        dmc = SQLLocation.active_objects.get_or_None(location_id=dmc_location_id)
+        if not dmc:
+            raise NikshayLocationNotFound(
+                "Location with id: {location_id} not found."
+                "This is the testing facility id assigned for test: {test_case_id}"
+                    .format(location_id=dmc_location_id, test_case_id=self.test_case.get_id)
+            )
+        nikshay_code = dmc.metadata.get('nikshay_code')
+        if not nikshay_code or (isinstance(nikshay_code, (str, unicode)) and not nikshay_code.isdigit()):
+            raise RequiredValueMissing("InAppt value for dmc, got value: {}".format(nikshay_code))
```

Comment:
Inappropriate. The code should be a string with just integer values. I think QA and test locations would keep nikshay code like "QA123" which eventually fails at Nikshay since it expects to parse it as integer.